core/src/worker/data-feed-l2.utils.ts

```diff
@@ -0,0 +1,38 @@
+import { IOracleRoundState } from '../types'
+import { ethers } from 'ethers'
+import { L2_PROVIDER_URL } from '../settings'
+import { Aggregator__factory } from '@bisonai/orakl-contracts'
+import { Logger } from 'pino'
+
+export async function oracleRoundStateCallL2({
```

Comment:
It would be much better if we don't have to "reinvent" the wheel. L1 Data Feed has similar function and the only difference seems to be L2_PROVIDER_URL. If we have no reason, I think we should use just one implementation, rather than two.

Additionally, it would be good to solve problems the same way everywhere. For example, the `oracleRoundStateCall` does not create a `JSONRpcProvider` but loads it from a settings.ts where it is generated once and reused in other places. If you think this approach is not good, we can make it better, but we should keep the solution for both L1 and L2 providers the same.

https://github.com/Bisonai/orakl/blob/51918df59402e458ee5d21ec754f42ed90c24c16/core/src/worker/data-feed.utils.ts#L51-L82

Reply:
Yes, this make sense. In this case we need to edit the `oracleRoundStateCall()` function, using RPC as a parameter.